node/index.js

```diff
@@ -367,6 +367,11 @@ TChannel.prototype.request = function channelRequest(options) {
     if (!options.service && self.serviceName) {
         options.service = self.serviceName;
     }
+
+    if (!self.topChannel && !options.host) {
```

Comment:
Let's shift this to be slightly different please:
- key on `!self.serviceName` not `!self.topChannel`
- rename the error to taste if you want to

Semantically, what you really mean is "you must specify a host if this is a generic cross-service channel", however this is a potential use case:
```
var channel = TChannel({
  serviceName: "foo"
});
```

In which case there are no top/sub channels, but the channel is explicitly single-serviced.

I'd like to try to have strict *bahn clients use this mode unless there's a compelling reason not to.